---
title: "rsubread"
author: "Tyler_Akonom"
date: "4/3/2023"
output: html_document
---

```{bash}
# Creating "interactive job" on alpine partition
sinteractive --partition=amilan --time=04:00:00 --ntasks=36

# Open and run R
source /curc/sw/anaconda/default
conda activate r361
R
```


```{r}
# Running Rsubread for counts:
library(Rsubread)

# CD to TMP

TMPDIR = "/gpfs/alpine1/scratch/tyak9569/"
TMP = "/gpfs/alpine1/scratch/tyak9569/"
TEMP = "/gpfs/alpine1/scratch/tyak9569/"

files=dir("/gpfs/alpine1/scratch/tyak9569/wbh/aligned", pattern = "*.bam", recursive = TRUE, full.names = TRUE)

# Point to GTF file:

gtf=("/gpfs/alpine1/scratch/tyak9569/wbh/genome/hg38.refGene.gtf")

# Output the file:

OUT="/gpfs/alpine1/scratch/tyak9569/wbh/rsub_out/"

# Unstranded

coverage=featureCounts(files,
isGTFAnnotationFile = TRUE,
annot.ext = gtf,
GTF.attrType = "gene_id",
allowMultiOverlap = TRUE,
nthreads = 36,
isPairedEnd=TRUE,
strandSpecific = 0)

fileroot=paste(OUT,"rsub_id", sep="")
write.csv(coverage$counts, paste(fileroot,".coverage.csv", sep=""))
write.csv(coverage$stat, paste(fileroot,".stat.csv", sep=""))
write.csv(coverage$annotation, paste(fileroot,".annotation.csv", sep=""))
write.csv(coverage$targets, paste(fileroot,".targets.csv", sep=""))
```
